hearth/lib/hearth/middleware/send.rb

```diff
@@ -79,37 +69,99 @@ def call(input, context)
 
       private
 
+      def stub_response(input, context, output)
+        stub = @stubs.next(context.operation_name)
+        apply_stub(stub, input, context, output)
+        return unless context.response.body.respond_to?(:rewind)
+
+        context.response.body.rewind
+      end
+
+      def send_request(context, output)
+        @client.transmit(
+          request: context.request,
+          response: context.response,
+          logger: context.logger
+        )
+      rescue Hearth::NetworkingError => e
+        output.error = e
+      end
+
       def apply_stub(stub, input, context, output)
         case stub
         when Proc
           stub = stub.call(input, context)
           apply_stub(stub, input, context, output) if stub
-        when Exception
+        when Exception, ApiError
           output.error = stub
-        when Class
-          output.error = stub.new
         when Hash
-          @stub_class.stub(
-            context.response,
-            stub: @params_class.build(
-              stub,
-              context: 'stub'
-            )
-          )
+          apply_stub_hash(stub, context)
         when NilClass
-          @stub_class.stub(
-            context.response,
-            stub: @params_class.build(
-              @stub_class.default,
-              context: 'stub'
-            )
-          )
+          apply_stub_nil(context)
+        when Hearth::Structure
```

Comment:
I think we may want to validate that the structure passed is actually the right one for the operation.  Though I realize we don't actually have that type here - but I think like we add PARAMS_CLASS to the stub, I think its reasonable to add the type class somewhere (either in the stub or in the params class).

Reply:
I added a TYPES_CLASS to data stubs.